central/cluster/datastore/datastore_impl_postgres_test.go

```diff
@@ -292,120 +292,143 @@ func (s *ClusterPostgresDataStoreTestSuite) TestLookupOrCreateClusterFromConfig(
 			}, &storage.SensorDeploymentIdentification{
 				AppNamespace:      "stackrox",
 				SystemNamespaceId: "456",
-			}),
+			}, nil),
 			shouldClusterBeUpserted: true,
 			shouldHaveLookupError:   true,
 		},
 		{
 			description: "try adding a cluster and then seeing if we get the same cluster back",
-			cluster:     someClusterWithManagerType(storage.ManagerType_MANAGER_TYPE_HELM_CHART, &someHelmConfig),
+			cluster:     someClusterWithManagerType(storage.ManagerType_MANAGER_TYPE_HELM_CHART, &someHelmConfig, nil),
 			sensorHello: sensorHelloWithHelmManagedConfigInit(&central.HelmManagedConfigInit{
 				ClusterConfig: &someHelmConfig,
 				ManagedBy:     storage.ManagerType_MANAGER_TYPE_HELM_CHART,
-			}, nil),
+			}, nil, nil),
 			shouldClusterBeUpserted: true,
 			expectedManagerType:     storage.ManagerType_MANAGER_TYPE_HELM_CHART,
 			expectedHelmConfig:      &someHelmConfig,
 		},
 		{
 			description: "try adding a cluster and then changing the manager type",
-			cluster:     someClusterWithManagerType(storage.ManagerType_MANAGER_TYPE_HELM_CHART, &someHelmConfig),
+			cluster:     someClusterWithManagerType(storage.ManagerType_MANAGER_TYPE_HELM_CHART, &someHelmConfig, nil),
 			sensorHello: sensorHelloWithHelmManagedConfigInit(&central.HelmManagedConfigInit{
 				ManagedBy: storage.ManagerType_MANAGER_TYPE_MANUAL,
-			}, nil),
+			}, nil, nil),
 			shouldClusterBeUpserted: true,
 			expectedManagerType:     storage.ManagerType_MANAGER_TYPE_MANUAL,
 			expectedHelmConfig:      nil,
 		},
 		{
 			description: "try adding a cluster and then changing the helm config",
-			cluster:     someClusterWithManagerType(storage.ManagerType_MANAGER_TYPE_HELM_CHART, &someHelmConfig),
+			cluster:     someClusterWithManagerType(storage.ManagerType_MANAGER_TYPE_HELM_CHART, &someHelmConfig, nil),
 			sensorHello: sensorHelloWithHelmManagedConfigInit(&central.HelmManagedConfigInit{
 				ManagedBy:     storage.ManagerType_MANAGER_TYPE_HELM_CHART,
 				ClusterConfig: &differentConfigFPHelmConfig,
-			}, nil),
+			}, nil, nil),
 			shouldClusterBeUpserted: true,
 			expectedManagerType:     storage.ManagerType_MANAGER_TYPE_HELM_CHART,
 			expectedHelmConfig:      &differentConfigFPHelmConfig,
 		},
+		{
+			description: "test that sensor capabilities match does not trigger an update",
+			cluster:     someClusterWithManagerType(storage.ManagerType_MANAGER_TYPE_HELM_CHART, &someHelmConfig, []string{"capability1", "capability2"}),
+			sensorHello: sensorHelloWithHelmManagedConfigInit(&central.HelmManagedConfigInit{
+				ClusterConfig: &someHelmConfig,
+				ManagedBy:     storage.ManagerType_MANAGER_TYPE_HELM_CHART,
+			}, nil, []string{"capability2", "capability1"}), // capability ordering should not matter
+			shouldClusterBeUpserted:    false,
+			expectedSensorCapabilities: []string{"capability1", "capability2"},
+		},
+		{
+			description: "test that sensor capabilities mismatch triggers an update",
```

Comment:
The capability sets in this case are completely disjoint. I think it would be more interesting for ~~me~~ someone  trying to deduce the intended behaviour from tests, if there was (additionally?) a case where the sets overlap partially (or a strict subset case).